src/credentials/providers/environmentCredentialsProvider.ts

```diff
@@ -0,0 +1,15 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { CredentialsProvider } from "./credentials";
+
+export interface EnvironmentCredentialsProvider extends CredentialsProvider {
+
+    /**
+     * Determines if the provider is currently capable of producing credentials.
+     */
+    isAvailable(): Promise<boolean>
```

Comment:
could put this directly on `CredentialsProvider` interface, instead of creating a subtype. I would even say that is preferable, because it avoids "death by specificity" (overuse of nominal types). Semantically, `isAvailable()` seems compatible with `CredentialsProvider`.

Also avoids the awkward distinction of `EnvironmentCredentialsProvider` vs `EnvVarsCredentialsProvider` (where the names are semantically identical but they are different programming constructs).

Reply:
Wouldn't worry about additive changes, and in our case even breaking changes don't matter if/until Toolkit is released as a public "library" / API. (unless such breaking changes are very disruptive to other pending PRs)